scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/KryoCoder.scala

```diff
@@ -0,0 +1,28 @@
+package com.twitter.scalding.beam_backend
+
+import com.esotericsoftware.kryo.io.Input
+import com.twitter.chill.{ KryoInstantiator, KryoPool }
+import java.io.{ InputStream, OutputStream }
+import org.apache.beam.sdk.coders.AtomicCoder
+import scala.language.implicitConversions
+
+final class KryoCoder(kryoInstantiator: KryoInstantiator) extends AtomicCoder[Any] {
+  @transient lazy val kryoPool = KryoPool.withByteArrayOutputStream(Runtime
+    .getRuntime
+    .availableProcessors, kryoInstantiator)
+  override def encode(value: Any, os: OutputStream): Unit = {
+    val bytes = kryoPool.toBytesWithClass(value)
+    os.write(bytes.length)
+    os.write(bytes)
+    os.flush()
+  }
+  override def decode(is: InputStream): Any = {
+    val size = is.read()
+    val input = new Input(is, size)
+    kryoPool.fromBytes(input.readBytes(size))
+  }
+}
+object KryoCoder {
+  implicit def inferType[T](atomicCoder: AtomicCoder[Any]): AtomicCoder[T] =
```

Comment:
1. can we not call this `inferType` but perhaps `castType`
2. can we accept a `KryoCoder` as the argument not an `AtomicCoder`? It isn't safe in general, it is safe(ish) for kryo because it can handle any type at runtime.